tail/tail.go

```diff
@@ -191,12 +191,22 @@ func tailSingleFile(tailer *tail.Tail, file string, stateFile string) chan strin
 	}()
 
 	go func() {
-		for line := range tailer.Lines {
-			if line.Err != nil {
-				// skip errored lines
-				continue
+	ReadLines:
+		for {
+			select {
+			case line, ok := <-tailer.Lines:
+				if !ok {
+					// tailer.Lines is closed
+					break ReadLines
+				}
+				if line.Err != nil {
+					// skip errored lines
+					continue
+				}
+				lines <- strings.TrimSpace(line.Text)
+			case <-abort:
+				break ReadLines
```

Comment:
🤔 how does ordering impact behavior here? seems like precedence here would let you not have to wait for a line to come in